src/txHashBuilder.c

```diff
@@ -1466,9 +1545,168 @@ static void txHashBuilder_assertCanLeaveNetworkId(tx_hash_builder_t* builder)
 	}
 }
 
-void txHashBuilder_finalize(tx_hash_builder_t* builder, uint8_t* outBuffer, size_t outSize)
+void txHashBuilder_enterCollateralReturn(tx_hash_builder_t *builder)
 {
+	_TRACE("state = %d", builder->state);
+
 	txHashBuilder_assertCanLeaveNetworkId(builder);
+	{
+		// Enter collateral output
+		BUILDER_APPEND_CBOR(CBOR_TYPE_UNSIGNED, TX_BODY_KEY_COLLATERAL_RETURN);
+		BUILDER_APPEND_CBOR(CBOR_TYPE_ARRAY, builder->includeCollateralReturn);
```

Comment:
This looks suspicious, why an array? Item 16 is a single element of type `transaction_output = legacy_transaction_output / post_alonzo_transaction_output`, so two possibilities.

I don't think we even need `txHashBuilder_enterCollateralReturn`, those `txHashBuilder_enter*` functions are for arrays/maps.